packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -219,20 +219,30 @@ export class TabBarRenderer extends TabBar.Renderer {
             }
             top = `${paddingTop + iconHeight}px`;
         }
-        const style: ElementInlineStyle = { width, height, top };
+
+        const style: ElementInlineStyle = {
+            color: this.getDecorationData(data.title, 'fontData')[0]?.color || 'unset',
+            height,
+            top,
+            width,
+        };
+
+        const tailDecorations = this.getTailDecorations(data.title);
+        const label = `${data.title.label}${tailDecorations !== '' ? ` ${tailDecorations}` : ''}`;
```

Comment:
It appears that in VSCode, the equivalent of `title.label` and `tailDecorations` can be styled separately, so maybe we should put the tail decorations in a separate element rather than joining them together as strings?